packages/sit-onyx/src/components/OnyxStepper/OnyxStepper.vue

```diff
@@ -0,0 +1,253 @@
+<script lang="ts" setup>
+import minus from "@sit-onyx/icons/minus.svg?raw";
+import plus from "@sit-onyx/icons/plus.svg?raw";
+import { computed, ref } from "vue";
+import { useDensity } from "../../composables/density";
+import { useCustomValidity } from "../../composables/useCustomValidity";
+import { injectI18n } from "../../i18n";
+import OnyxFormElement from "../OnyxFormElement/OnyxFormElement.vue";
+import OnyxIcon from "../OnyxIcon/OnyxIcon.vue";
+import OnyxLoadingIndicator from "../OnyxLoadingIndicator/OnyxLoadingIndicator.vue";
+import OnyxSkeleton from "../OnyxSkeleton/OnyxSkeleton.vue";
+import type { OnyxStepperProps } from "./types";
+
+const props = withDefaults(defineProps<OnyxStepperProps>(), {
+  modelValue: 0,
+  step: 1,
+  stripStep: false,
+  placeholder: "0",
+  disabled: false,
+  readonly: false,
+  loading: false,
+  skeleton: false,
+});
+
+const { t } = injectI18n();
+const inputRef = ref<HTMLInputElement>();
+
+const emit = defineEmits<{
+  /** Emitted when the input value changes. */
+  "update:modelValue": [value: number];
+  /**
+   * Emitted when the input is focussed.
+   */
+  focus: [];
+  /**
+   * Emitted when the input is blurred.
+   */
+  blur: [];
+  /**
+   * Emitted when the validity state of the input changes.
+   */
+  validityChange: [validity: ValidityState];
+}>();
+
+const { vCustomValidity, errorMessages } = useCustomValidity({ props, emit });
+
+const { densityClass } = useDensity(props);
+
+/**
+ * Current value (with getter and setter) that can be used as "v-model" for the native input.
+ */
+const value = computed({
+  get: () => props.modelValue,
+  set: (value) => emit("update:modelValue", value),
+});
+
+const handleClick = (direction: "up" | "down") => {
+  if (!inputRef.value) return;
+
+  switch (direction) {
+    case "down":
+      inputRef.value.stepDown();
+      break;
+    case "up":
+      inputRef.value.stepUp();
+  }
+
+  emit("update:modelValue", inputRef.value.valueAsNumber);
+};
```

Comment:
This could be shortened like this:
```suggestion
const handleClick = (direction: "stepDown" | "stepUp") => {
  if (!inputRef.value) return;
  inputRef.value[direction]();
  emit("update:modelValue", inputRef.value.valueAsNumber);
};
```